roles/openshift_logging_elasticsearch/tasks/main.yaml

```diff
@@ -450,6 +459,57 @@
     es_deploy_name: "{{ openshift_logging_elasticsearch_deployment_name }}"
   when: openshift_logging_elasticsearch_deployment_name != ""
 
+- set_fact:
+    _es_container_security_context: {'privileged': true}
+  when:
+  - openshift_logging_elasticsearch_storage_type == "hostmount"
+
+- set_fact:
+    _es_container_security_context: "{{ _es_containers.elasticsearch.securityContext }}"
+  when:
+  - _es_containers is defined
+  - '"elasticsearch" in _es_containers'
+  - '"securityContext" in _es_containers.elasticsearch'
+
+- set_fact:
+    _es_node_selector: "{{ openshift_logging_elasticsearch_nodeselector | default({}) }}"
+  when:
+  - openshift_logging_elasticsearch_storage_type != "hostmount"
+
+# label nodes for hostmount path
+- when:
+  - openshift_logging_elasticsearch_storage_type == "hostmount"
+  block:
+  - set_fact:
+      _es_node_selector: "{{ openshift_logging_elasticsearch_hostmount_nodeselector }}"
+
+  - name: check if a node has already been allocated for dc
+    oc_obj:
+      state: list
+      kind: node
+      selector: "{{ _es_node_selector | map_to_pairs }}"
+    register: es_hostmount_selected_node
+
+  - when:
+    - "es_hostmount_selected_node['results']['results'][0]['items'] | default([]) | length == 0"
+    block:
+    - name: Retrieve list of schedulable es nodes that are not already allocated for another dc
+      oc_obj:
+        state: list
+        kind: node
+        selector: "{{ openshift_logging_elasticsearch_nodeselector | map_to_pairs }},{{ openshift_logging_elasticsearch_hostmount_nodeselector_key }} notin ({% for iter in range( openshift_logging_es_cluster_size | int ) -%}{{ iter }}{% if not loop.last %},{% endif %}{% endfor -%})"
+        field_selector: "spec.unschedulable!=true"
+      register: es_available_nodes
+
```

Comment:
```suggestion

    - name: "Ensure that has nodes for ES to run on"
      assert:
        that: "{{  es_available_nodes['results']['results'][0]['items'] | default([]) | length != 0 }}"
        msg: |-
          No schedulable nodes found matching node selector for elastic search.'

```

Reply:
@vrutkovs assertion to guarantee schedulable nodes as you suggested.

Reply:
Ah, excellent, thanks!